### PR TITLE
Run Prettier and fix ESLint complaints 

### PR DESCRIPTION
**Pull request by alexnguyenn**
As title

### CODE REVIEW

**Review by gwwatkin — APPROVED** [review]

**Comment by alexnguyenn** [comment]
@gwwatkin I turned off `@typescript-eslint/ban-ts-comment` which interferes with `@ts-ignore` comment in `BaseTemplate.tsx`. If you know how to fix it feel free to do so. 

Also here are the last warnings I am having for LatticeView.tsx:
```
157:35  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
157:35  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
163:34  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
163:34  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
```
Since they are warnings they won't fail any CI we setup later on, but if you can fix it go ahead.

**Comment by gwwatkin** [comment]
Get this error:
```
> ls-ui-react-app@0.1.0 lint:fix
> eslint --fix src/**/*.{js,jsx,ts,tsx}


Oops! Something went wrong! :(

ESLint: 7.32.0

No files matching the pattern "src/**/*.js" were found.
Please check for typing mistakes in the pattern.
```
Which is strange because, `npx eslint --fix src/**/*.{js,jsx,ts,tsx}` runs as expected

**Comment by gwwatkin** [comment]
Ok found the problem the `npx eslint --fix ./src/**/*.{js,jsx,ts,tsx}` also doesn't work, its `npx eslint --fix ./src/**/*.{ts,tsx}` that works

**Comment by gwwatkin** [comment]
Tried this: https://github.com/import-js/eslint-plugin-import/issues/1645#issuecomment-595808778
also doesn't work

**Comment by gwwatkin** [comment]
This is the only thing that fixes it, though it excludes js and jsx files from liniting
```diff
diff --git a/package.json b/package.json
index 2828b95..8353fac 100644
--- a/package.json
+++ b/package.json
@@ -34,8 +34,8 @@
     "build": "react-scripts build",
     "test": "react-scripts test",
     "eject": "react-scripts eject",
-    "lint": "eslint src/**/*.{js,jsx,ts,tsx}",
-    "lint:fix": "eslint --fix src/**/*.{js,jsx,ts,tsx}"
+    "lint": "eslint src/**/*.{ts,tsx}",
+    "lint:fix": "eslint --fix src/**/*.{ts,tsx}"
   },
   "browserslist": {
     "production": [

```

**Comment by gwwatkin** [comment]
Not that it's a big problem to excluding js and jsx from linting, but it would be nice to combine it with something that forbids `js` and `jsx`. Tried the `"allowJs": false,` option in `tsconfig.json` but it also forbids importing js which is going to be a problem with modules like JQuery.

Also if we do end up adding some plain js we can always just re enable the option.

File: package.json [hunk 1]
```diff
@@ -34,8 +34,8 @@
     "build": "react-scripts build",
     "test": "react-scripts test",
     "eject": "react-scripts eject",
-    "lint": "eslint src/**/*.{js,jsx,ts,tsx}",
-    "lint:fix": "eslint --fix src/**/*.{js,jsx,ts,tsx}"
+    "lint": "eslint --no-error-on-unmatched-pattern src/**/*.{js,jsx,ts,tsx}",
+    "lint:fix": "eslint --no-error-on-unmatched-pattern --fix src/**/*.{js,jsx,ts,tsx}"
```
Comment (gwwatkin) [inline package.json:38]: Ok it turns out I could just add this option here, sorry for the noise in the PR

**Review by gwwatkin — APPROVED** [review]
Added some fixes
- `--no-error-on-unmatched-pattern`
- null assertion warnings
- removed the ts-ignore directive

File: src/pages/BaseTemplate.tsx [hunk 2]
```diff
@@ -9,8 +9,8 @@ import UploadCircuitPage from "./UploadCircuitPage"
 import { AppState } from "../appState"
 
 const anchor_position_shift = (window: Window) => {
-    // @ts-ignore
-    if ($(window).width() > 991) {
+    const windowWidth = $(window).width() ?? 1000 // Default width
+    if (windowWidth > 991) {
```
Comment (gwwatkin) [inline src/pages/BaseTemplate.tsx:13]: @Keelando any idea if `.width()` can ever return null?

File: package.json (hunk 1 above)
Comment (alexnguyenn) [reply package.json:38]: no problem, nice to see that we have a fix for this.

**Pull request merged**